### ISSUE TITLE
2008 Multnomah general has corruption

### ISSUE THREAD

**Issue by nk9** [openelections/openelections-data-or#141]
I haven't figured out what's going on, but `20081104__or__general__multnomah__precinct.csv` has all sorts of issues. An excerpt, from line 4050:

```
Multnomah,Vote for,U.S. Senate,,REP,Gordon H Smith,1
Multnomah,Vote for,U.S. Senate,,IND,Joel Haugen,1
Multnomah,Vote for,U.S. Senate,,PAC,Michael Meo,1
Multnomah,0156 156,U.S. Senate,,REP,Gordon H Smith,43
Multnomah,0156 156,U.S. Senate,,IND,Joel Haugen,43
Multnomah,0156 156,U.S. Senate,,PAC,Michael Meo,43
Multnomah,0156 156,U.S. Senate,,DEM,Kurt Schrader,43
Multnomah,0156 156,U.S. Senate,,DEM,Jeff Merkley,18
Multnomah,0156 156,U.S. Senate,,PAC,Chris Henry,18
Multnomah,0156 156,U.S. Senate,,DEM,Earl Blumenauer,18
Multnomah,0156 156,U.S. Senate,,REP,Mike Erickson,18
Multnomah,0156 156,U.S. Senate,,CON,Dave Brownlow,0
Multnomah,0156 156,U.S. Senate,,CON,Scott Semrau,0
Multnomah,0156 156,U.S. Senate,,REP,Delia Lopez,0
Multnomah,0156 156,U.S. Senate,,LIB,Steve Milligan,0
```

Earl Blumenauer is a US Rep!

**Comment by dwillis** [comment]
Sounds like someone (me) was an overzealous copier.

**Comment by nk9** [comment]
I think we've all been guilty of that at one time or another!

**Comment by dwillis** [comment]
Fixed now.

**Comment by nk9** [comment]
I've added a check to the verifier which ensures that a given line is unique, except for the vote count. This _should_ be true; if it's not, then there could be issues later on in the pipeline when seemingly the same contest shows multiple vote counts.

The Multnomah 2006 results fail this new test, and in ways I can't quite understand. But it seems similar to what was happening with the 2008 results. E.g. one of the errors is that on line 1447, Pete Sorenson is supposed to have received 94 votes for Governor in precinct 1505. But he didn't, he received 18 as properly recorded on line 853. And Betsy Johnson was running for State Senator, not Governor.

This looks like an algorithm error, so hopefully it will be easy to fix.

**Comment by nk9** [comment]
Here's the current state of play on the Multnomah corruption. Four csvs have duplicate data. One has some invalid columns. (Should I be less strict about that in the verifier?)

```
⋊> ~/P/o/OR-data on master ⨯ python src/verifier.py --singleError */*multnomah__precinct.csv                                                                                                 01:45:18
==> 2004/20040518__or__primary__multnomah__precinct.csv
==> 2004/20041102__or__general__multnomah__precinct.csv
==> 2006/20060516__or__primary__multnomah__precinct.csv
ERROR: Line 1447: Line is duplicated (original line 853)
{'votes': '94', 'district': '', 'candidate': 'Pete Sorenson', 'office': 'Governor', 'county': 'Multnomah', 'party': 'Democratic', 'precinct': '1505'}
==> 2006/20061107__or__general__multnomah__precinct.csv
ERROR: Line 1833: Line is duplicated (original line 1041)
{'votes': '145', 'district': '', 'candidate': 'Joe Keating', 'office': 'Governor', 'county': 'Multnomah', 'party': 'PAC', 'precinct': '1505'}
==> 2008/20080520__or__primary__multnomah__precinct.csv
==> 2008/20081104__or__general__multnomah__precinct.csv
==> 2010/20100518__or__primary__multnomah__precinct.csv
ERROR: Line 782: Line is duplicated (original line 8)
{'votes': '2', 'district': '', 'candidate': 'Pavel Goberman', 'office': 'U.S. Senate', 'county': 'Multnomah', 'party': 'Democrat', 'precinct': '0158 158'}
==> 2010/20101102__or__general__multnomah__precinct.csv
ERROR: Line 1042: Line is duplicated (original line 10)
{'votes': '0', 'district': '', 'candidate': 'Jim Huffman', 'office': 'U.S. Senate', 'county': 'Multnomah', 'party': 'REP', 'precinct': '0158 158'}
==> 2012/20120515__or__primary__multnomah__precinct.csv
==> 2012/20121106__or__general__multnomah__precinct.csv
==> 2014/20140520__or__primary__multnomah__precinct.csv
==> 2014/20141104__or__general__multnomah__precinct.csv
==> 2016/20160517__or__primary__multnomah__precinct.csv
ERROR: Line 0: Invalid columns: set(['votes_for_others', 'over_votes_with_candidate', 'under_votes_without_candidate', 'total_ballots'])
==> 2016/20161108__or__general__multnomah__precinct.csv
```

**Comment by nk9** [comment]
Resolved with pull request #155
